index.html

```diff
@@ -2866,32 +2850,52 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
         W3C TAG recommendations</a> to avoid using URNs in the Web.
         Therefore, Web NFC does not use the URN prefix neither when reading
-        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
-        before the type name.
+        or writing <a>NDEF messages</a>.
       </p>
       <p>
-        To <dfn>validate external type</dfn> given a |input:USVString| run
+        To <dfn>validate external type</dfn> given |input:USVString|, run
         these steps:
       </p>
       <ol class=algorithm id="validate-external-type">
         <li>
           If |input:string| is not a {{USVString}} or it is empty, or its
-          length exceeds 255 bytes, or it contains white spaces,
+          length exceeds 255 bytes, return `false`.
+        </li>
+        <li>
+          Find the last occurrence of colon (`':'`) in |input|. If not found,
           return `false`.
         </li>
         <li>
-          If |input| does not start with colon ('`:`'), return |input|.
-          <p class="issue">
-            Validating domain name needs an algorithm.
-          </p>
+          Otherwise let |domain| be the part of |input| preceding the last colon,
+          and |type| be the part following the last colon.
+        </li>
+        <li>
+          Run the following sub-steps of <a href="">host parsing</a> on |domain|:
+          <ul>
+            <li>
+              Let |asciiDomain| be the result of running
+              <a href="https://url.spec.whatwg.org/#concept-domain-to-ascii">
+              domain to ASCII</a> on |domain|.
+            </li>
+            <li>
+              If |asciiDomain| is failure, return `false`.
+            </li>
+            <li>
+              If |asciiDomain| contains a
+              <a href="https://url.spec.whatwg.org/#forbidden-host-code-point">
+              forbidden host code point</a>, return `false`.
+            </li>
+          </ul>
+        </li>
+        <li>
+          If |type| is empty, return `false`.
         </li>
         <li>
-          If |input| starts with colon ('`:`'), let |domain:string| be the
-          <a data-cite="url#concept-url-serializer">serialization</a> of the
-          <a data-cite="html">effective domain</a> of the {{Document/origin}}.
+          If |type| does not match the `custom-type` definition from the
+          <a>external type name ABNF</a>, return `false`.
         </li>
         <li>
-          Return |domain| concatenated with |input|.
+          Return |asciiDomain| concatenated with `':'` and |type|.
```

Comment:
Are we interested in this, or just return `true` and let recordType be the same that was provided by the app?

Reply:
Nah, we can readd then